### PR TITLE
[4.4] Fix cancel frontend module editing redirection

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue #42862.

### Summary of Changes
Currently, if you edit a module from frontend of your site and decide to cancel editing, you will always be redirected to homepage instead of to the page which you clicked on Edit button. This PR fixes that wrong behavior

### Testing Instructions
1. Use Joomla 4.4
2. Login to frontend of your site using a super user account
3. Access to certain page (not homepage) and try to click on edit button to edit certain module
4. On edit screen, click on Cancel button to cancel editing

### Actual result BEFORE applying this Pull Request
You are being redirected to homepage of your site

### Expected result AFTER applying this Pull Request
You are being redirected to original page which you clicked on Edit button to start editing.

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 88cbca23f537e3bea461886ed315c7a09135e9e6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42870">issues.joomla.org/tracker/joomla-cms/42870</a>.</sub>

File: components/com_config/src/Controller/ModulesController.php [hunk 1]
```diff
@@ -59,8 +59,8 @@ public function __construct($config = [], MVCFactoryInterface $factory = null, $
      */
     public function cancel()
     {
-        // Redirect back to home(base) page
-        $this->setRedirect(Uri::base());
+        // Redirect back to preview page
```
Comment (brianteeman) [inline components/com_config/src/Controller/ModulesController.php]: "preview"? did you mean "previous"

File: components/com_config/src/Controller/ModulesController.php [hunk 2]
```diff
@@ -150,19 +150,29 @@ public function save()
...
-                }
-
-                $this->setRedirect($redirect);
+                $this->setRedirect($this->getReturnUrl());
                 break;
         }
     }
+
+    /**
+     * Method to get redirect URL after saving or cancel editing a module fromfrontend
```
Comment (brianteeman) [inline components/com_config/src/Controller/ModulesController.php]:
```suggestion
     * Method to get redirect URL after saving or cancel editing a module from frontend
```

File: components/com_config/src/Controller/ModulesController.php (hunk 1 above)
Comment (joomdonation) [reply components/com_config/src/Controller/ModulesController.php]: Yes, I meant previous. Thanks Brian.

**Comment by faustonenci** [comment]
I have tested this item :white_check_mark: successfully on 583c3764100344a6d9e664d2b6534c10538bc0b3<br /><br />good work<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42870">issues.joomla.org/tracker/joomla-cms/42870</a>.</sub>

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on 583c3764100344a6d9e664d2b6534c10538bc0b3<br /><br />On J 5.1 remotely it works<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42870">issues.joomla.org/tracker/joomla-cms/42870</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42870">issues.joomla.org/tracker/joomla-cms/42870</a>.</sub>

File: components/com_config/src/Controller/ModulesController.php [hunk 3]
```diff
@@ -150,19 +150,29 @@ public function save()
...
                 break;
         }
     }
+
+    /**
+     * Method to get redirect URL after saving or cancel editing a module from frontend
+     *
+     * @return  string
+     *
+     * @since   __DEPLOY_VERSION_
```
Comment (heelc29) [inline components/com_config/src/Controller/ModulesController.php]:
```suggestion
     * @since   __DEPLOY_VERSION__
```

**Pull request merged**